File: scripts/sdk/ingest.py
#!/usr/bin/env python3
"""Minimal ThermalTrace ingest SDK."""
import json
import os
import sys
import urllib.request

base = (sys.argv[1] if len(sys.argv) > 1 else None) or os.environ.get("THERMALTRACE_URL", "https://thermaltrace.dev")
key = (sys.argv[2] if len(sys.argv) > 2 else None) or os.environ.get("THERMALTRACE_INGEST_KEY")
temp = float((sys.argv[3] if len(sys.argv) > 3 else None) or os.environ.get("TEMP", "42"))

if not key:
    print("Usage: ingest.py [baseUrl] <deviceKey> [tempF]", file=sys.stderr)
    sys.exit(1)

url = f"{base.rstrip('/')}/api/ingest/{key}"
req = urllib.request.Request(
    url,
    data=json.dumps({"temp1": temp}).encode(),
    headers={"Content-Type": "application/json"},
    method="POST",
)
with urllib.request.urlopen(req) as resp:
    print(resp.status, resp.read().decode())

File: scripts/sdk/ingest.mjs
#!/usr/bin/env node
/** Minimal ThermalTrace ingest SDK — Node 18+ */
const baseUrl = process.argv[2] || process.env.THERMALTRACE_URL || "https://thermaltrace.dev";
const deviceKey = process.argv[3] || process.env.THERMALTRACE_INGEST_KEY;
const temp = Number(process.argv[4] ?? process.env.TEMP ?? "42");

if (!deviceKey) {
  console.error("Usage: node ingest.mjs [baseUrl] <deviceKey> [tempF]");
  process.exit(1);
}

const res = await fetch(`${baseUrl.replace(/\/$/, "")}/api/ingest/${deviceKey}`, {
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify({ temp1: temp }),
});

console.log(res.status, await res.text());
